### ISSUE TITLE
rspamd-mailcow Container is unhealthy

### ISSUE THREAD

**Issue by skupfer** [mailcow/mailcow-dockerized#114]
Clean git pull, clean build.

Unfortunately I cannot bring up the rspamd-container...

How do I view the logs? "docker logs container" doesn't show any errors

**Comment by andryyy** [comment]
It takes a few seconds for it to come up.

What do you mean with "doesn't come up" exactly?

**Comment by skupfer** [comment]
There is no container for rspamd-mailcow

```
ERROR: for rspamd-mailcow  Container "542eac87ca11" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```

```
compose.parallel.feed_queue: Healthcheck for service(s) upstream of <Service: rspamd-mailcow> failed - not processing
compose.parallel.parallel_execute_iter: Failed: <Service: rspamd-mailcow>
compose.parallel.feed_queue: Pending: set([])
```

**Comment by andryyy** [comment]
Can you pull the image and not build it? Try to remove all images and run docker pull mailcow/rspamd please

**Comment by andryyy** [comment]
I am also rebuilding it now.

**Comment by skupfer** [comment]
Just removed all images, used a single docker pull rspamd and let it run. Was okayish I think. After that I pulled the user images and used docker-compose to bring them up (as per installation guide)... 

I am still receiving the same error... Is there any way I can provide you a more detailed log/error details?

EDIT: Quick try in my VM, same iptables-rules, same system. Working there... How do I find out what the problem is?

EDIT2:

```
ERROR: for rspamd-mailcow  Container "82772373072f" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```
which is the nginx container id... followed by
```
docker inspect --format "{{json .State.Health }}" 82772373072f | jq '.Log[].Output'
"PING php-fpm-mailcow (172.22.1.3): 56 data bytes\n--- php-fpm-mailcow ping statistics ---\n10 packets transmitted, 0 packets received, 100% packet loss\n"
"PING php-fpm-mailcow (172.22.1.3): 56 data bytes\n--- php-fpm-mailcow ping statistics ---\n10 packets transmitted, 0 packets received, 100% packet loss\n"
"PING php-fpm-mailcow (172.22.1.3): 56 data bytes\n--- php-fpm-mailcow ping statistics ---\n10 packets transmitted, 0 packets received, 100% packet loss\n"
"PING php-fpm-mailcow (172.22.1.3): 56 data bytes\n--- php-fpm-mailcow ping statistics ---\n10 packets transmitted, 0 packets received, 100% packet loss\n"
"PING php-fpm-mailcow (172.22.1.3): 56 data bytes\n--- php-fpm-mailcow ping statistics ---\n10 packets transmitted, 0 packets received, 100% packet loss\n"
```

EDIT3: Ok, different approach. Used "docker-compose run -d rspamd-mailcow" to the container up -> mailcowdockerized_rspamd-mailcow_run_1

This is the log from the container
[log.json.txt](https://github.com/andryyy/mailcow-dockerized/files/841063/log.json.txt)

EDIT4: I get the following log entry or the rspamd-container. Surely enough this is working for most users. Tried to find the problem but didn't succeed yet. Working perfectly fine within my virtual machine as I said already...

```
{"log":"container_linux.go:247: starting container process caused \"exec: \\\"-d\\\": executable file not found in $PATH\"\n","stream":"stderr","time":"2017-03-14T09:38:13.133315872Z"}
```

**Comment by andryyy** [comment]
Can you show us the docker version?

**Comment by skupfer** [comment]
Sure. My VM is using the same

docker version
```
Client:
 Version:      17.03.0-ce
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   60ccb22
 Built:        Thu Feb 23 10:53:29 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.0-ce
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   60ccb22
 Built:        Thu Feb 23 10:53:29 2017
 OS/Arch:      linux/amd64
 Experimental: false
```
docker-compose version
```
docker-compose version 1.11.2, build dfed245
docker-py version: 2.1.0
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.1t  3 May 2016
```

**Comment by paladine-raistlin** [comment]
The container that is unhealthy is nginx. See below. Fresh install. Thank You. 
```

/mailcow-dockerized# docker-compose up -d
mailcowdockerized_mysql-mailcow_1 is up-to-date
mailcowdockerized_bind9-mailcow_1 is up-to-date
mailcowdockerized_rmilter-mailcow_1 is up-to-date
mailcowdockerized_postfix-mailcow_1 is up-to-date
mailcowdockerized_redis-mailcow_1 is up-to-date
mailcowdockerized_memcached-mailcow_1 is up-to-date
mailcowdockerized_dovecot-mailcow_1 is up-to-date
mailcowdockerized_php-fpm-mailcow_1 is up-to-date
mailcowdockerized_sogo-mailcow_1 is up-to-date
mailcowdockerized_nginx-mailcow_1 is up-to-date

ERROR: for rspamd-mailcow  Container "5c5a2261916c" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```

```
/mailcow-dockerized# docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                          PORTS                                                                                                                               NAMES
ebbd41c5cbc0        mailcow/rspamd      "/bin/bash -c ' sl..."   42 minutes ago      Up 3 minutes                    11333-11334/tcp                                                                                                                     mailcowdockerized_rspamd-mailcow_1
5c5a2261916c        nginx:mainline      "/bin/bash -c 'env..."   43 minutes ago      Restarting (1) 18 seconds ago                                                                                                                                       mailcowdockerized_nginx-mailcow_1
93e7881184fd        mailcow/sogo        "/bin/sh -c 'exec ..."   43 minutes ago      Up 3 minutes                                                                                                                                                        mailcowdockerized_sogo-mailcow_1
980ea421f1e6        mailcow/dovecot     "/docker-entrypoin..."   43 minutes ago      Up 3 minutes                    0.0.0.0:110->110/tcp, 0.0.0.0:143->143/tcp, 0.0.0.0:993->993/tcp, 0.0.0.0:995->995/tcp, 24/tcp, 10001/tcp, 0.0.0.0:4190->4190/tcp   mailcowdockerized_dovecot-mailcow_1
84da7521b8b9        mailcow/phpfpm      "/docker-entrypoin..."   43 minutes ago      Up 3 minutes                    9000/tcp                                                                                                                            mailcowdockerized_php-fpm-mailcow_1
dbfad1e56f89        memcached           "docker-entrypoint..."   43 minutes ago      Up 3 minutes                    11211/tcp                                                                                                                           mailcowdockerized_memcached-mailcow_1
c4bf812e2dd6        mailcow/postfix     "/bin/sh -c 'exec ..."   43 minutes ago      Up 3 minutes                    0.0.0.0:25->25/tcp, 0.0.0.0:465->465/tcp, 0.0.0.0:587->587/tcp, 588/tcp                                                             mailcowdockerized_postfix-mailcow_1
b0fa284bc0d2        redis               "docker-entrypoint..."   43 minutes ago      Up 3 minutes                    6379/tcp                                                                                                                            mailcowdockerized_redis-mailcow_1
0da524d76c5a        mailcow/rmilter     "/bin/sh -c 'exec ..."   43 minutes ago      Up 3 minutes                    9000/tcp                                                                                                                            mailcowdockerized_rmilter-mailcow_1
9f66010175c3        resystit/bind9      "named -c /etc/bin..."   43 minutes ago      Up 3 minutes                    53/tcp                                                                                                                              mailcowdockerized_bind9-mailcow_1
00c95e742f34        mariadb:10.1        "docker-entrypoint..."   44 minutes ago      Up 3 minutes (healthy)          3306/tcp                                                                                                                            mailcowdockerized_mysql-mailcow_1
```

**Comment by paladine-raistlin** [comment]
Update.
figured out problem and got it working
issue is nginx needs a hardcopy of `data/assets/ssl/cert.pem` and `data/assets/ssl/key.pem` a hard link to these files is not possible if one has the folder `mailcow-dockerized` on a different zpool/zfs or disk than that of /etc so I did `ln -s` which is no good. A `cp of cert and key` works but is not practical for obvious reasons. Will attempt to mount a zfs to` /etc/letsencrypt` and see if that works. 
The reason to put `mailcow-dockerized` on same zpool as docker is ease of restore to different server in case of failure. If someone has a suggestion that would be better, am all ears.

**Comment by SISheogorath** [comment]
If you setup the let's encrypt renewal like I did in the following blog post: https://www.shivering-isles.com/lets-encrypt-renew-all-your-certs-using-systemd/

You can use the same way I use there to reload the apache config to copy the TLS certificate into the zfs pool and restart nginx.

**Comment by skupfer** [comment]
I am not on zfs

```
stat -f -c %T /
ext2/ext3
```

Trying to figure out why nginx container is unable to ping php-fpm-mailcow ... There is a 100% packet loss

EDIT: found something ...

```
sudo netstat -lnp | grep 172
udp        0      0 172.17.0.1:53725        0.0.0.0:*                           582/btsync-daemon
udp        0      0 172.22.1.1:44700        0.0.0.0:*                           582/btsync-daemon
udp        0      0 172.22.1.1:35545        0.0.0.0:*                           582/btsync-daemon
udp        0      0 172.22.1.1:123          0.0.0.0:*                           507/ntpd
udp        0      0 172.17.0.1:123          0.0.0.0:*                           507/ntpd
```

**Comment by skupfer** [comment]
Alright, definitely a problem with the network. Other applications/services were blocking 172.22.*!
Changed every line I found using 172.22. to 172.25. and the containers are all coming up healthy and working!

Maybe you could use a different network by default. I guess it's a common ip :-)

EDIT: only weird thing. I haven't opened a single port in iptables and the mails are coming through :-) But whatever, working so far and I am happy. This is my achievement today *it's something*

Thanks for your support, all of you!

**Comment by skupfer** [comment]
Forgot to close. 
After some kernel panics (see https://forum.mailcow.email/t/mailcow-docker-causes-kernel-panic-edit/448/6) I updated to the most recent kernel available for jessie-backports and as of now its seems to run stable. I also updated the repo and repulled everthing. I am not sure how and why but now it is also working for the "standard network" (was before updating the kernel due to restarts)
